webui/react/src/ee/SamlAuth.ts

```diff
@@ -3,6 +3,11 @@ import router from 'router';
 export const samlUrl = (basePath: string, queries?: string): string => {
   if (queries) {
     queries = queries.replace(/r=0\.\d+[&]?/, '');
+    if (queries.includes('&redirect=')) {
+      const redirectStartIndex = queries.indexOf('&redirect=');
+      const redirectEndIndex = queries.indexOf('&', redirectStartIndex + 1);
```

Comment:
does this work if `&redirect=` is the end of the URL, and `redirectEndIndex` is -1 ?

Reply:
@mapmeld updated and added test cases